CMakeLists.txt

```diff
@@ -1,95 +1,75 @@
+endif()
+
+set(definitions "UNDERSCORE" "NFILES=32" "MAXCD=250" "MAXNC=600" "MXNAF=3")
+add_compile_definitions(definitions)
+
+if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
+  include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Intel.cmake)
   set(IntelComp true )
-elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU*" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang*")
+elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
   set(GNUComp true )
-elseif(CMAKE_CXX_COMPILER_ID MATCHES "pgc*")
-  set(PGIComp true )
+  include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/GNU.cmake)
+  set(allocation_def "DYNAMIC_ALLOCATION")
 endif()
 
+set(fortran_8_flags ${FREAL8} ${FINT8} ${TRACEBACK})
+set(shared_flags  ${DEBINFO} ${FOPT3} ${AVX2})
+set(fortran_8_DA_flags ${FREAL8} ${FINT8} ${TRACEBACK})
+set(fortran_d_DA_flags ${FREAL8} ${TRACEBACK})
+set(c_SS_flags ${MMMED} ${SHARED})
+
 STRING(COMPARE EQUAL ${CMAKE_BUILD_TYPE} "RelWithDebInfo" BUILD_RELEASE)
 STRING(COMPARE EQUAL ${CMAKE_BUILD_TYPE} "RELEASE" BUILD_RELEASE)
 STRING(COMPARE EQUAL ${CMAKE_BUILD_TYPE} "PRODUCTION" BUILD_PRODUCTION)
 STRING(COMPARE EQUAL ${CMAKE_BUILD_TYPE} "DEBUG" BUILD_DEBUG)
 
 option(GSIBUILD "Build only libs for GSI" ON)
 
-set(definitions "UNDERSCORE" "NFILES=32" "MAXCD=250" "MAXNC=600" "MXNAF=3")
-
-if( (BUILD_RELEASE) OR (BUILD_PRODUCTION) )
-  if(IntelComp)
-    #shared with C and Fortran
-    set(shared_flags "-g" "-traceback" "-O3" "-axCORE-AVX2")
-
-    set(c_DA_allocation_def "DYNAMIC_ALLOCATION")
-    set(c_4_DA_flags)    
-    set(c_8_DA_definitions "F77_INTSIZE_8")
-    set(c_d_DA_flags)
-    
-    set(fortran_4_DA_flags)
-    set(fortran_8_DA_flags "-r8" "-i8")
-    set(fortran_d_DA_flags "-r8")
-    
-    set(c_nonDA_allocation_def "STATIC_ALLOCATION")
-    set(c_4_flags)
-    set(c_8_flags)
-    set(c_8_definitions "F77_INTSIZE_8")
-    set(c_d_flags)
-    set(c_SS_flags "-mcmodel=medium" "-shared")
-
-    set(fortran_4_flags)
-    set(fortran_8_flags "-r8" "-i8")
-    set(fortran_d_flags "-r8")
-    set(fortran_SS_flags "-mcmodel=medium" "-shared")
-
-  elseif(GNUComp)
-    set(allocation_def "DYNAMIC_ALLOCATION")
-    set(shared_flags "-O3" "-ggdb" "-Wall")
-
-    set(c_d_flags)
-    set(c_4_flags)
-    set(c_8_flags)
-
-    set(shared_fortran_flags "-funroll-loops")
-    set(fortran_4_flags)
-    set(fortran_d_flags "-fdefault-real-8")
-    set(fortran_8_flags "-fdefault-integer-8" "-fdefault-real-8")
-  else()
-    message("unknown compiler!")
-    exit()
-  endif()
-endif()
+set(c_DA_allocation_def "DYNAMIC_ALLOCATION")
+set(c_nonDA_allocation_def "STATIC_ALLOCATION")
 
+set(c_4_DA_flags "-DUNDERSCORE")
+set(c_8_DA_definitions "F77_INTSIZE_8")
+set(c_d_DA_flags "-DUNDERSCORE")
+set(fortran_4_DA_flags)
+set(c_4_flags  "-DUNDERSCORE")
+set(c_8_flags  "-DUNDERSCORE")
+set(c_8_definitions "F77_INTSIZE_8")
+set(c_d_flags "-DUNDERSCORE")
+set(fortran_4_flags)
 
 file(GLOB fortran_src ${CMAKE_CURRENT_SOURCE_DIR}/src/*.f ${CMAKE_CURRENT_SOURCE_DIR}/src/*.F)
```

Comment:
Please do not glob. and move the building of the library into src.
The top level CMakeLists.txt should not do anything more than basic setup and add_subdirectory.